app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -269,74 +269,77 @@ def index_for_group_type
+    if [UserGroup.group_types[:councils], UserGroup.group_types[:teams_committees]].include?(group_type)
+      status = params.require(:status)
+      already_existing_member = TeamMember.find_by(team_id: original_group_id, user_id: user_id, end_date: nil)
+      if already_existing_member.present?
+        already_existing_member.update!(end_date: Date.today)
+      end
+      TeamMember.create!(team_id: original_group_id, user_id: user_id, start_date: Date.today, team_leader: status == "leader", team_senior_member: status == "senior_member")
+      render json: {
+        success: true,
+      }
+    else
+      render status: :unprocessable_entity, json: { error: "Invalid group type" }
+    end
+  end
+
   def create
     user_id = params.require(:userId)
     group_id = params[:groupId] || UserGroup.find_by(group_type: params.require(:groupType)).id
 
-    if group_id.is_a?(String) && group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
-      # migrated to the new system.
-      group_type = group_id_of_old_system_to_group_type(group_id)
-      original_group_id = group_id.split("_").last
-      return head :unauthorized unless current_user.can_edit_team?(original_group_id)
-      if [UserGroup.group_types[:councils], UserGroup.group_types[:teams_committees]].include?(group_type)
-        status = params.require(:status)
-        already_existing_member = TeamMember.find_by(team_id: original_group_id, user_id: user_id, end_date: nil)
-        if already_existing_member.present?
-          already_existing_member.update!(end_date: Date.today)
-        end
-        TeamMember.create!(team_id: original_group_id, user_id: user_id, start_date: Date.today, team_leader: status == "leader", team_senior_member: status == "senior_member")
-        render json: {
-          success: true,
-        }
-      else
-        render status: :unprocessable_entity, json: { error: "Invalid group type" }
-      end
-    elsif group.group_type == UserGroup.group_types[:delegate_regions] && !delegate_status_migrated?(status)
+    if team_role?(group_id)
+      return create_team_role(group_id, user_id)
+    end
+
+    create_supported_groups = [
+      UserGroup.group_types[:delegate_regions],
+      UserGroup.group_types[:delegate_probation],
+      UserGroup.group_types[:translators],
+    ]
+    group = UserGroup.find(group_id)
+    status = params.require(:status) if UserGroup.group_types_containing_status_metadata.include?(group.group_type)
+    location = params[:location] if group.group_type == UserGroup.group_types[:delegate_regions]
+
+    return render status: :unprocessable_entity, json: { error: "Invalid group type" } unless create_supported_groups.include?(group.group_type)
+    return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group_id)
+
+    if status.present? && group.unique_status?(status)
+      end_role_for_user_in_group_with_status(group, status)
```

Comment:
Should the controller return here? Otherwise the flow of the method will continue to create a new user role below